core.coffee

```diff
@@ -434,6 +447,35 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				data = utils.cloneDeep(data)
```

Comment:
I think for most we can get first aim to get rid of the usage, eg I don't see why we need to `cloneDeep here`, and the `.map` below can be replaced with a coffeescript for loop, then we can look at what to do for any that are remaining (adding extra requirements is definitely an option, but I'd like them to be options that are commonly available so that providing them is easy for every backend - it also makes it a backwards incompatible change)